src/types.rs

```diff
@@ -1534,17 +1757,100 @@ impl Rule {
             None
         }
     }
-    /// Get all the Variables in the Rule.
+    /// Get all the [`Variable`]s in the `Rule`.
+    ///
+    /// [`Variable`]: struct.Variable.html
+    /// 
+    /// # Examples
+    /// 
+    /// ```
+    /// # use term_rewriting::{Signature, Term, Rule, parse_term, parse_rule};
+    /// let mut sig = Signature::default();
+    /// 
+    /// let r = parse_rule(&mut sig, "A(x_) = C(x_)").expect("parse of A(x_) = C(x_)");
+    /// let r2 = parse_rule(&mut sig, "B(y_ z_) = C").expect("parse of B(y_ z_) = C");
+    /// 
+    /// let r_variables: Vec<String> = r.variables().iter().map(|v| v.display(&sig)).collect();
+    /// let r2_variables: Vec<String> = r2.variables().iter().map(|v| v.display(&sig)).collect();
+    ///
+    /// assert_eq!(r_variables, vec!["x_"]);
+    /// assert_eq!(r2_variables, vec!["y_", "z_"]);
+    /// ```
     pub fn variables(&self) -> Vec<Variable> {
         self.lhs.variables()
     }
-    /// Get all the Operators in the Rule.
+    /// Get all the [`Operator`]s in the `Rule`.
+    ///
+    /// [`Operator`]: struct.Operator.html
+    /// # Examples
+    /// 
+    /// ```
+    /// # use term_rewriting::{Signature, Term, Rule, parse_rule};
+    /// let mut sig = Signature::default();
+    /// 
+    /// let r = parse_rule(&mut sig, "A(D E) = C").expect("parse of A(D E) = C");
+    /// let r2 = parse_rule(&mut sig, "B(F x_) = C").expect("parse of B(F x_) = C");
+    /// 
+    /// let r_ops: Vec<String> = r.operators().iter().map(|o| o.display(&sig)).collect();
+    /// let r2_ops: Vec<String> = r2.operators().iter().map(|o| o.display(&sig)).collect();
+    ///
+    /// assert_eq!(r_ops, vec!["D", "E", "A", "C"]);
+    /// assert_eq!(r2_ops, vec!["F", "B", "C"]);
+    /// ```
     pub fn operators(&self) -> Vec<Operator> {
         let lhs = self.lhs.operators().into_iter();
         let rhs = self.rhs.iter().flat_map(Term::operators);
         lhs.chain(rhs).unique().collect()
     }
-    /// Get all the subterms and places in a Rule.
+    /// Get all the subterms and places in a `Rule`.
+    ///
+    /// See [`Term`] for more information.
+    ///
+    /// [`Term`]: enum.Term.html
+    ///
+    /// # Examples
+    /// 
+    /// ```
+    /// # use term_rewriting::{Signature, Term, Rule, parse_rule};
+    /// let mut sig = Signature::default();
+    /// 
+    /// let r = parse_rule(&mut sig, "A = C | D").expect("parse of A = C | D");
+    /// 
+    /// let subterms: Vec<String> = r.subterms()
+    ///     .iter()
+    ///     .map(|(i, t, p)| format!("{},Subterm:{},Place:{:?}", i, t.display(&sig), p))
+    ///     .collect();
+    /// 
+    /// assert_eq!(
+    ///     subterms,
+    ///     vec![
+    ///         "0,Subterm:A,Place:[]",
+    ///         "1,Subterm:C,Place:[]",
+    ///         "2,Subterm:D,Place:[]",
+    ///     ]
+    /// );
```

Comment:
I think the second example is more informative than the first. Do you any specific reason to keep the first example? If not, please remove.